static/js/lib/ckeditor/plugins/SiteContentEmbed.ts

```diff
@@ -41,6 +45,35 @@ class SiteContentMarkdownSyntax extends MarkdownSyntaxPlugin {
   }
 }
 
+/**
+ * A CKEditor Command for inserting a new SiteContentEmbed (siteContent)
+ * node into the editor.
+ */
+class InsertSiteContentEmbedCommand extends Command {
```

Comment:
I'm not exactly sure where to put this comment, but I think there is an issue with rendering the shortcode in the markdown output.  I created a test site with a resource and embedded it in a page, and then published.  This is the markdown output I got for that page:

```
---
title: Test Page
type: page
uid: 65048845-2e79-4346-93ed-2d06a31df9a5
---
Test content

5b3431f6-c62e-4257-a1e2-644271dfa65d
```
It seems to just be putting the raw UID in there, whereas it should be surrounded in a shortcode.  I think the strategy of using a generic "resource" shortcode is a good idea, although we may want to think about the naming concerns I noted in one of my other comments.  I think maybe we should be a bit more generic about which collection contains the content to embed, since it's named with `name: resource` in the course template and might not always be that way.  Maybe we could do something like `{{< content-embed ${name} ${content} >}}`, that way Hugo knows what `type` of content to look for, and then can fetch the right one by UID.  Since the resource itself contains the rest of the data in its front matter, from there we can fetch the `filetype` and have conditional templates for each type of content defined.

As far as the types of resources, since those are just currently arbitrary strings defined in the config file ("Image, Video, Document, Other"), we should explicitly document which types of resources we support embedding in the `ocw-hugo-themes`' documentation when we get to that part.  I imagine for "Other" we would use some kind of file download fallback tempalte with just a download button, and we could also use that when a `filetype` is defined that doesn't match one supported by the theme.  Either way, we should make sure it's wrapped in some kind of shortcode, otherwise the UID will just appear as text.

Reply:
oh hmm I'm not sure what's going on there - could you check the storage format too?

Reply:
When I add a resource in the editor and then check the saved content in the API, I see the shortcodes I expect:

![Screen Shot 2021-08-06 at 2 55 40 PM](https://user-images.githubusercontent.com/6207644/128558725-96d2533b-2ce7-43bf-b5f0-971c21d93de3.png)

so maybe we have an issue on the pipeline rendering part stripping the shortcodes out somehow?

Reply:
The resulting `WebsiteContent` object created in the database on my instance did not have the shortcode:

```

Reply:
I re-tested this with the latest changes and tried creating a new Page and everything worked:

```
HTTP 200 OK
Allow: GET, PUT, PATCH, DELETE, HEAD, OPTIONS
Content-Type: application/json
Vary: Accept

{
    "text_id": "1d7acff5-215d-40f5-94d8-c71c8c35774f",
    "type": "page",
    "content_context": null,
    "title": "Test Page 2",
    "markdown": "Test content 2\n\n{{< resource 5b3431f6-c62e-4257-a1e2-644271dfa65d >}}",
    "metadata": null,
    "file": null
}
```